x-pack/plugins/apm/server/lib/service_map/get_service_map_service_node_info.ts

```diff
@@ -30,69 +33,72 @@ interface Options {
 }
 
 interface TaskParameters {
-  setup: Setup;
-  minutes: number;
+  environment?: string;
   filter: ESFilter[];
+  minutes: number;
+  serviceName?: string;
+  setup: Setup;
 }
 
 export async function getServiceMapServiceNodeInfo({
   serviceName,
-  environment,
   setup,
-}: Options & { serviceName: string; environment?: string }) {
+  uiFilters,
+}: Options & { serviceName: string; uiFilters: UIFilters }) {
   const { start, end } = setup;
 
   const filter: ESFilter[] = [
     { range: rangeFilter(start, end) },
     { term: { [SERVICE_NAME]: serviceName } },
-    ...getEnvironmentUiFilterES(environment),
+    ...getEnvironmentUiFilterES(uiFilters.environment),
   ];
 
   const minutes = Math.abs((end - start) / (1000 * 60));
-  const taskParams = { setup, minutes, filter };
+  const taskParams = {
+    environment: uiFilters.environment,
+    filter,
+    minutes,
+    serviceName,
+    setup,
+  };
 
   const [
-    errorMetrics,
+    errorStats,
     transactionStats,
-    cpuMetrics,
-    memoryMetrics,
+    cpuStats,
+    memoryStats,
   ] = await Promise.all([
-    getErrorMetrics(taskParams),
+    getErrorStats(taskParams),
     getTransactionStats(taskParams),
-    getCpuMetrics(taskParams),
-    getMemoryMetrics(taskParams),
+    getCpuStats(taskParams),
+    getMemoryStats(taskParams),
   ]);
-
   return {
-    ...errorMetrics,
+    ...errorStats,
     transactionStats,
-    ...cpuMetrics,
-    ...memoryMetrics,
+    ...cpuStats,
+    ...memoryStats,
   };
 }
 
-async function getErrorMetrics({ setup, minutes, filter }: TaskParameters) {
-  const { client, indices } = setup;
-
-  const response = await client.search({
-    index: indices['apm_oss.errorIndices'],
-    body: {
-      size: 0,
-      query: {
-        bool: {
-          filter: filter.concat({ term: { [PROCESSOR_EVENT]: 'error' } }),
-        },
-      },
-      track_total_hits: true,
-    },
-  });
-
-  return {
-    avgErrorsPerMinute:
-      response.hits.total.value > 0
-        ? response.hits.total.value / minutes
-        : null,
+async function getErrorStats({
+  setup,
+  serviceName,
+  environment,
+}: {
+  setup: Options['setup'];
+  serviceName: string;
+  environment?: string;
+}) {
+  const setupWithBlankUiFilters = {
+    ...setup,
+    uiFiltersES: getEnvironmentUiFilterES(environment),
   };
```

Comment:
i'm not sure `setupWithBlankUiFilters` is required: it could be redundant with `setup.uiFiltersES`. which should be part of the setup object (typed with `SetupUIFilters`).

Reply:
I think it is @ogupte , since `setup.uiFiltersES` has the `service.name` of the route. And it's not what you want on Service Maps, you want the service.name from the node selected, right? So If we don't remove it, we'll have two service.name filters one from the route and another from the node.